# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/boxbackup/hotfix-const.patch
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

Make the set comparators const, required by newer libstdc++.

--- a/lib/backupstore/HousekeepStoreAccount.cpp	2020-03-26 21:07:58.000000000 +0100
+++ b/lib/backupstore/HousekeepStoreAccount.cpp	2026-09-27 19:16:39.333937715 +0200
@@ -561,7 +561,7 @@ bool HousekeepStoreAccount::ScanDirector
 //		Created: 11/12/03
 //
 // --------------------------------------------------------------------------
-bool HousekeepStoreAccount::DelEnCompare::operator()(const HousekeepStoreAccount::DelEn &x, const HousekeepStoreAccount::DelEn &y)
+bool HousekeepStoreAccount::DelEnCompare::operator()(const HousekeepStoreAccount::DelEn &x, const HousekeepStoreAccount::DelEn &y) const
 {
 	// STL spec says this:
 	// A Strict Weak Ordering is a Binary Predicate that compares two objects, returning true if the first precedes the second.
--- a/lib/backupstore/HousekeepStoreAccount.h	2020-03-26 21:07:58.000000000 +0100
+++ b/lib/backupstore/HousekeepStoreAccount.h	2026-09-27 19:16:39.328838668 +0200
@@ -72,7 +72,7 @@ private:
 	
 	struct DelEnCompare
 	{
-		bool operator()(const DelEn &x, const DelEn &y);
+		bool operator()(const DelEn &x, const DelEn &y) const;
 	};
 	
 	int mAccountID;
--- a/lib/bbackupd/BackupDaemon.cpp	2020-03-26 21:07:58.000000000 +0100
+++ b/lib/bbackupd/BackupDaemon.cpp	2026-09-27 19:16:39.324227354 +0200
@@ -2339,7 +2339,7 @@ void BackupDaemon::SendSyncStartOrFinish
 	// by code, rather than the OS.
 	typedef struct
 	{
-		bool operator()(const std::string &s1, const std::string &s2)
+		bool operator()(const std::string &s1, const std::string &s2) const
 		{
 			if(s1.size() == s2.size())
 			{
